iroh-bytes/src/util.rs

```diff
@@ -156,7 +160,13 @@ impl Serialize for Hash {
     where
         S: Serializer,
     {
-        serializer.serialize_bytes(self.0.as_bytes())
+        // Fixed-length structures, including arrays, are supported in Serde as tuples
+        // See: https://serde.rs/impl-serialize.html#serializing-a-tuple
+        let mut s = serializer.serialize_tuple(32)?;
+        for item in self.0.as_bytes() {
```

Comment:
this seems horribly inefficient. But I guess it is fine since in our bulk data path we don't use this.

Reply:
There is no other way, at least not in serde: 

https://docs.rs/serde/latest/src/serde/ser/impls.rs.html#163-168